File: src/InfraLLM.Core/Models/Job.cs
using InfraLLM.Core.Enums;

namespace InfraLLM.Core.Models;

public class Job
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Prompt { get; set; }
    public JobTriggerType TriggerType { get; set; }
    public string? CronSchedule { get; set; }
    public string? WebhookSecret { get; set; }
    public bool AutoRunLlm { get; set; }
    public bool IsEnabled { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? LastRunAt { get; set; }

    public Organization Organization { get; set; } = null!;
    public List<JobRun> Runs { get; set; } = [];
}

File: frontend/src/types/job.ts
export type JobTriggerType = "Cron" | "Webhook";

export interface Job {
  id: string;
  organizationId: string;
  userId: string;
  name: string;
  description: string | null;
  prompt: string | null;
  triggerType: JobTriggerType;
  cronSchedule: string | null;
  webhookSecret: string | null;
  autoRunLlm: boolean;
  isEnabled: boolean;
  createdAt: string;
  updatedAt: string;
  lastRunAt: string | null;
}

export interface JobRun {
  id: string;
  jobId: string;
  sessionId: string | null;
  triggeredBy: string;
  status: string;
  payload: string;
  response: string | null;
  createdAt: string;
  completedAt: string | null;
}

export interface CreateJobRequest {
  name: string;
  description?: string;
  prompt?: string;
  triggerType: JobTriggerType;
  cronSchedule?: string;
  webhookSecret?: string;
  autoRunLlm?: boolean;
  isEnabled?: boolean;
}

export interface UpdateJobRequest {
  name?: string;
  description?: string;
  prompt?: string;
  triggerType?: JobTriggerType;
  cronSchedule?: string;
  webhookSecret?: string;
  autoRunLlm?: boolean;
  isEnabled?: boolean;
}
